hooks/boot/04-services.rb

```diff
@@ -0,0 +1,44 @@
+require 'open3'
+
+module ServicesHookContextExtension
+  ALL_POSSIBLE_SERVICES = [
+    '*mongo*', # Used by Pulp 2
+    '*redis*', # Used by Foreman/Dynflow and Pulpcore
+    'apache.service', # Apache on Debian
+    'dynflow*',
+    'foreman*',
+    'httpd.service', # Apache on Red Hat
+    'postgresql*', # Used by Foreman/Dynflow and Pulpcore
+    'pulp*',
+    'qdrouterd.service', # Used by Pulp 2
+    'qpidd.service', # Used by Pulp 2
+    'smart_proxy_dynflow_core.service', # Used by Foreman Proxy
+    'tomcat.service', # Candlepin
+    'squid.service', # Used by Pulp 2
+  ].freeze
+
+  def start_services(services)
```

Comment:
Can this default to ALL_POSSIBLE_SERVICES similar to stop_srevices?

Reply:
I see this doesn't handle well from systemd:

```
Failed to start services: Failed to start apache.service: Unit not found.
Failed to start smart_proxy_dynflow_core.service: Unit not found.
[ERROR 2020-07-01T14:57:11 verbose] Failed to start services: Failed to start apache.service: Unit not found.
Failed to start smart_proxy_dynflow_core.service: Unit not found.
```

So while I feel it should be possible to just start everything, forcing a developer to define what to start is OK too given this response from systemd.

Reply:
> Can this default to ALL_POSSIBLE_SERVICES similar to stop_srevices?

IMHO no since this is mostly used to start services prior to a migration. In that case you only want to start really what's needed.
```
$ rg start_services
spec/services_hook_extensions_spec.rb
15:  context 'start_services' do
16:    subject { context.start_services(services) }

hooks/boot/04-services.rb
20:  def start_services(services)

hooks/pre/30-el7_upgrade_postgresql.rb
2:  start_services(['postgresql'])

katello/hooks/pre/10-reset_feature.rb
8:    start_services(['postgresql'])
```